go/vt/vttablet/tabletmanager/tm_state.go

```diff
@@ -181,7 +181,23 @@ func (ts *tmState) ChangeTabletType(ctx context.Context, tabletType topodatapb.T
 		// Update the tablet record first.
 		_, err := topotools.ChangeType(ctx, ts.tm.TopoServer, ts.tm.tabletAlias, tabletType, PrimaryTermStartTime)
 		if err != nil {
-			return err
+			log.Errorf("Error changing type in topo record for tablet %s :- %v\nWill keep trying to read from the toposerver", topoproto.TabletAliasString(ts.tm.tabletAlias), err)
+			// In case of a topo error, we aren't sure if the data has been written or not.
+			// We must read the data again and verify whether the previous write succeeded or not.
+			// The only way to guarantee safety is to keep retrying read until we succeed
+			for {
+				ti, errInReading := ts.tm.TopoServer.GetTablet(ctx, ts.tm.tabletAlias)
+				if errInReading != nil {
+					<-time.After(100 * time.Millisecond)
```

Comment:
This is even more compact than my original suggestion. +1

Reply:
💕